### PR TITLE
Enable MacOS M1 stages in the CI on a weekly basis

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Run build/test in the CI on `macos m1` workers. It will run on a weekly basis

## Why is it important?

More test coverage

## Issues

There are test failures for MacOS M1:

* [ ] `Auditbeat` - https://github.com/elastic/beats/issues/33033
* [ ] `Metricbeat` - https://github.com/elastic/beats/issues/33035
* [ ] `x-pack/metricbeat` - https://github.com/elastic/beats/issues/33036

@cmacknz @kpollich , how would you like to proceed? This particular implementation was discussed in an internal issue - you can find it linked to this particular PR with the name `configure the required projects to use the new Orka M1 ephemeral workers`

### CODE REVIEW

**Comment by v1v** [comment]
Terraform for MacOS is not available in the existing version (`0.13.7`) but from https://github.com/hashicorp/terraform/releases/tag/v1.0.2 
see https://releases.hashicorp.com/terraform/1.0.2/

Blocked by the bump

**Review by kuisathaverat — APPROVED** [review]

File: filebeat/Jenkinsfile.yml [hunk 1]
```diff
@@ -49,6 +49,19 @@ stages:
                 - "macosTest"
             tags: true         ## for all the tags
         stage: extended
+    macosM1:
+        mage: "mage build unitTest"
+        platforms:             ## override default label in this specific stage.
+          - "orka && darwin && aarch64"
+        when:                  ## Override the top-level when.
+            comments:
+                - "/test auditbeat for macos-m1"
```
Comment (belimawr) [inline filebeat/Jenkinsfile.yml:58]: I believe this should be Filebeat:
```suggestion
                - "/test filebeat for macos-m1"
```

File: heartbeat/Jenkinsfile.yml [hunk 2]
```diff
@@ -51,6 +51,19 @@ stages:
                 - "macosTest"
             tags: true         ## for all the tags
         stage: extended
+    macosM1:
+        mage: "mage build unitTest"
+        platforms:             ## override default label in this specific stage.
+          - "orka && darwin && aarch64"
+        when:                  ## Override the top-level when.
+            comments:
+                - "/test auditbeat for macos-m1"
```
Comment (belimawr) [inline heartbeat/Jenkinsfile.yml:60]:
```suggestion
                - "/test heartbeat for macos-m1"
```

File: metricbeat/Jenkinsfile.yml [hunk 3]
```diff
@@ -41,6 +41,19 @@ stages:
                 - "macosTest"
             tags: true         ## for all the tags
         stage: extended
+    macosM1:
+        mage: "mage build unitTest"
+        platforms:             ## override default label in this specific stage.
+          - "orka && darwin && aarch64"
+        when:                  ## Override the top-level when.
+            comments:
+                - "/test auditbeat for macos-m1"
```
Comment (belimawr) [inline metricbeat/Jenkinsfile.yml:50]:
```suggestion
                - "/test metricbeat for macos-m1"
```

File: packetbeat/Jenkinsfile.yml [hunk 4]
```diff
@@ -43,6 +43,19 @@ stages:
                 - "macosTest"
             tags: true         ## for all the tags
         stage: extended
+    macosM1:
+        mage: "mage build unitTest"
+        platforms:             ## override default label in this specific stage.
+          - "orka && darwin && aarch64"
+        when:                  ## Override the top-level when.
+            comments:
+                - "/test auditbeat for macos-m1"
```
Comment (belimawr) [inline packetbeat/Jenkinsfile.yml:52]:
```suggestion
                - "/test packetbeat for macos-m1"
```

File: x-pack/filebeat/Jenkinsfile.yml [hunk 5]
```diff
@@ -49,6 +49,19 @@ stages:
                 - "macosTest"
             tags: true         ## for all the tags
         stage: extended
+    macosM1:
+        mage: "mage build unitTest"
+        platforms:             ## override default label in this specific stage.
+          - "orka && darwin && aarch64"
+        when:                  ## Override the top-level when.
+            comments:
+                - "/test auditbeat for macos-m1"
```
Comment (belimawr) [inline x-pack/filebeat/Jenkinsfile.yml:58]:
```suggestion
                - "/test filebeat for macos-m1"
```

File: x-pack/functionbeat/Jenkinsfile.yml [hunk 6]
```diff
@@ -44,6 +44,19 @@ stages:
                 - "macosTest"
             tags: true         ## for all the tags
         stage: extended
+    macosM1:
+        mage: "mage build unitTest"
+        platforms:             ## override default label in this specific stage.
+          - "orka && darwin && aarch64"
+        when:                  ## Override the top-level when.
+            comments:
+                - "/test auditbeat for macos-m1"
```
Comment (belimawr) [inline x-pack/functionbeat/Jenkinsfile.yml:53]:
```suggestion
                - "/test functonbeat for macos-m1"
```

File: x-pack/heartbeat/Jenkinsfile.yml [hunk 7]
```diff
@@ -34,6 +34,19 @@ stages:
                 - "macosTest"
             tags: true         ## for all the tags
         stage: extended
+    macosM1:
+        mage: "mage build unitTest"
+        platforms:             ## override default label in this specific stage.
+          - "orka && darwin && aarch64"
+        when:                  ## Override the top-level when.
+            comments:
+                - "/test auditbeat for macos-m1"
```
Comment (belimawr) [inline x-pack/heartbeat/Jenkinsfile.yml:43]:
```suggestion
                - "/test heartbeat for macos-m1"
```

File: x-pack/metricbeat/Jenkinsfile.yml [hunk 8]
```diff
@@ -62,6 +62,19 @@ stages:
                 - "macosTest"
             tags: true         ## for all the tags
         stage: extended
+    macosM1:
+        mage: "mage build unitTest"
+        platforms:             ## override default label in this specific stage.
+          - "orka && darwin && aarch64"
+        when:                  ## Override the top-level when.
+            comments:
+                - "/test auditbeat for macos-m1"
```
Comment (belimawr) [inline x-pack/metricbeat/Jenkinsfile.yml:71]:
```suggestion
                - "/test metricbeat for macos-m1"
```

File: x-pack/osquerybeat/Jenkinsfile.yml [hunk 9]
```diff
@@ -32,6 +32,19 @@ stages:
                 - "macosTest"
             tags: true         ## for all the tags
         stage: extended
+    macosM1:
+        mage: "mage build unitTest"
+        platforms:             ## override default label in this specific stage.
+          - "orka && darwin && aarch64"
+        when:                  ## Override the top-level when.
+            comments:
+                - "/test auditbeat for macos-m1"
```
Comment (belimawr) [inline x-pack/osquerybeat/Jenkinsfile.yml:41]:
```suggestion
                - "/test osquerybeat for macos-m1"
```

File: x-pack/packetbeat/Jenkinsfile.yml [hunk 10]
```diff
@@ -47,6 +47,19 @@ stages:
                 - "macosTest"
             tags: true         ## for all the tags
         stage: extended
+    macosM1:
+        mage: "mage build unitTest"
+        platforms:             ## override default label in this specific stage.
+          - "orka && darwin && aarch64"
+        when:                  ## Override the top-level when.
+            comments:
+                - "/test auditbeat for macos-m1"
```
Comment (belimawr) [inline x-pack/packetbeat/Jenkinsfile.yml:56]:
```suggestion
                - "/test packetbeat for macos-m1"
```

**Review by belimawr — CHANGES_REQUESTED** [review]
I'm far from being an expert on our automation, but it seems the "comments" rule should match the Beat name.

File: auditbeat/Jenkinsfile.yml [hunk 11]
```diff
@@ -46,6 +46,19 @@ stages:
...
+          - "orka && darwin && aarch64"
+        when:                  ## Override the top-level when.
+            comments:
+                - "/test auditbeat for macos-m1"
+            labels:
+                - "macos-m1"
+            parameters:
+                - "macosM1Test"
+            tags: false         ## for all the tags
+        stage: extended
```
Comment (v1v) [inline auditbeat/Jenkinsfile.yml:61]:
```suggestion
      # Ignore as long as there are test failures, see https://github.com/elastic/beats/issues/33033
#     macosM1:
#         mage: "mage build unitTest"
#         platforms:             ## override default label in this specific stage.
#           - "orka && darwin && aarch64"
#         when:                  ## Override the top-level when.
#             comments:
#                 - "/test auditbeat for macos-m1"
#             labels:
#                 - "macos-m1"
#             parameters:
#                 - "macosM1Test"
#             tags: false         ## for all the tags
#         stage: extended
```

File: metricbeat/Jenkinsfile.yml [hunk 12]
```diff
@@ -41,6 +41,19 @@ stages:
...
+          - "orka && darwin && aarch64"
+        when:                  ## Override the top-level when.
+            comments:
+                - "/test metricbeat for macos-m1"
+            labels:
+                - "macos-m1"
+            parameters:
+                - "macosM1Test"
+            tags: false         ## for all the tags
+        stage: extended
```
Comment (v1v) [inline metricbeat/Jenkinsfile.yml:56]:
```suggestion
      # Ignore as long as there are test failures, see https://github.com/elastic/beats/issues/33035
#     macosM1:
#         mage: "mage build unitTest"
#         platforms:             ## override default label in this specific stage.
#           - "orka && darwin && aarch64"
#         when:                  ## Override the top-level when.
#             comments:
#                 - "/test metricbeat for macos-m1"
#             labels:
#                 - "macos-m1"
#             parameters:
#                 - "macosM1Test"
#             tags: false         ## for all the tags
#         stage: extended
```

File: x-pack/metricbeat/Jenkinsfile.yml [hunk 13]
```diff
@@ -62,6 +62,19 @@ stages:
...
+          - "orka && darwin && aarch64"
+        when:                  ## Override the top-level when.
+            comments:
+                - "/test metricbeat for macos-m1"
+            labels:
+                - "macos-m1"
+            parameters:
+                - "macosM1Test"
+            tags: false         ## for all the tags
+        stage: extended
```
Comment (v1v) [inline x-pack/metricbeat/Jenkinsfile.yml:77]:
```suggestion
      # Ignore as long as there are test failures, see https://github.com/elastic/beats/issues/33036
#     macosM1:
#         mage: "mage build unitTest"
#         platforms:             ## override default label in this specific stage.
#           - "orka && darwin && aarch64"
#         when:                  ## Override the top-level when.
#             comments:
#                 - "/test metricbeat for macos-m1"
#             labels:
#                 - "macos-m1"
#             parameters:
#                 - "macosM1Test"
#             tags: false         ## for all the tags
#         stage: extended
```

**Pull request merged**